command/transfer_coin.go

```diff
@@ -0,0 +1,87 @@
+package command
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/hex337/alex-koin-go/config"
+	"github.com/hex337/alex-koin-go/model"
+
+	"log"
+	"regexp"
+)
+
+type TransferCoinCommand struct{}
+
+func (c *TransferCoinCommand) Run(msg string, event *CoinEvent) (string, error) {
+	re := regexp.MustCompile(`^(?i)transfer (?P<amount>[0-9]+) (?:to )?\<@(?P<to_slack_id>[0-9A-Z]+)\> (?:for)??(?P<reason>.+)`)
+	matches := re.FindStringSubmatch(event.Message)
+
+	if len(matches) < 4 {
+		return "Invalid transfer format. Expected something like `@Alex Koin transfer 3 to @alexk for being amazing`. See the channel details for command syntax.", nil
+	}
+
+	amount, err := strconv.Atoi(matches[1])
+	if err != nil {
+		log.Printf("INF amount not parsed as int: %s", matches[1])
+		return "Invalid transfer amount.", nil
+	}
+	toUserId := matches[2]
+	reason := matches[3]
+
+	log.Printf("INF transfer coin amt: %d, toUserId: %s, reason: %s", amount, toUserId, reason)
+
+	toUser, err := model.GetOrCreateUserBySlackID(toUserId)
+
+	if err != nil {
+		log.Printf("Could not find user with slack id %s: %s", toUserId, err.Error())
+		return "", err
+	}
+
+	canTransfer, msg := canTransferCoin(event.User, amount)
+
+	if !canTransfer {
+		return msg, nil
+	}
+
+	var coinIds []int
```

Comment:
Could probably do this as a sub query, but this also works. I guess this introduces a race condition if someone tries to transfer a bunch of stuff really fast.